pontoon/pretranslation/tests/test_pretranslate.py

```diff
@@ -145,51 +145,152 @@ def test_get_pretranslations_fluent_empty(
+    pretranslated_string = serializer.serialize_entry(parser.parse_entry(expected))
+
+    response = get_pretranslations(fluent_entity, google_translate_locale)
+    assert response == [(pretranslated_string, None, tm_user)]
+
+
 @patch("pontoon.pretranslation.pretranslate.get_google_translate_data")
 @pytest.mark.django_db
 def test_get_pretranslations_fluent_complex(
     gt_mock, entity_a, fluent_resource, google_translate_locale, tm_user
 ):
     # Entity.string is a complex Fluent string.
-    # Only translate text nodes.
-    # Service with most translated nodes is picked as translation author.
+    # - Uplift selector and repeat shared parts within variants.
+    # - Only translate text nodes.
+    # - Service with most translated nodes is picked as translation author.
     gt_mock.return_value = {
         "status": True,
-        "translation": "gt_translation",
+        "translation": "GT: Open Options and select Search.",
     }
 
     TranslationMemoryFactory.create(
         entity=entity_a,
-        source="Preferences",
-        target="tm_translation",
+        source="Open Preferences and select Search.",
+        target="TM: Open Preferences and select Search.",
         locale=google_translate_locale,
     )
     TranslationMemoryFactory.create(
         entity=entity_a,
-        source="Settings",
-        target="tm_translation",
+        source="Open Settings and select Search.",
+        target="TM: Open Settings and select Search.",
         locale=google_translate_locale,
     )
 
     fluent_string = dedent(
         """
-        my-entry =
-            { PLATFORM() ->
+        complex-entry =
+            Open { PLATFORM() ->
                 [mac] Preferences
                 [win] Settings
                 *[other] Options
-            }
+            } and select Search.
     """
     )
     fluent_entity = EntityFactory(resource=fluent_resource, string=fluent_string)
 
     expected = dedent(
         """
-        my-entry =
+        complex-entry =
             { PLATFORM() ->
-                [mac] tm_translation
-                [win] tm_translation
-                *[other] gt_translation
+                [mac] TM: Open Preferences and select Search.
+                [win] TM: Open Settings and select Search.
+                *[other] GT: Open Options and select Search.
+            }
+    """
+    )
+
+    # Re-serialize to match whitespace
+    pretranslated_string = serializer.serialize_entry(parser.parse_entry(expected))
+
+    response = get_pretranslations(fluent_entity, google_translate_locale)
+    assert response == [(pretranslated_string, None, tm_user)]
+
+
+@pytest.mark.django_db
+def test_get_pretranslations_fluent_sibling_selectors(
+    entity_a, fluent_resource, google_translate_locale, tm_user
+):
+    # Entity.string is a Fluent string with two sibling selectors.
+    #
+    TranslationMemoryFactory.create(
+        entity=entity_a,
+        source="{ $key_count } key and ",
```

Comment:
We could remove the whitespace here if we strip whitespace from `node.value` before it gets passed to `get_pretranslated_data()`. Is there a scenario in which this would get back at us?

Reply:
If we have significant leading or trailing whitespace, it sounds pretty risky to trust the automated systems to keep it intact. The current flattener doesn't handle that for non-selector strings, though.